cli/src/cmd/forge/coverage.rs

```diff
@@ -117,110 +124,132 @@ impl CoverageArgs {
         Ok((project, output))
     }
 
-    /// Builds the coverage map.
-    fn prepare(&self, output: ProjectCompileOutput) -> eyre::Result<(CoverageMap, SourceMaps)> {
+    /// Builds the coverage report.
+    fn prepare(
+        &self,
+        config: &Config,
+        output: ProjectCompileOutput,
+    ) -> eyre::Result<CoverageReport> {
+        let project_paths = config.project_paths();
+
         // Extract artifacts
         let (artifacts, sources) = output.into_artifacts_with_sources();
-        let artifacts: HashMap<ArtifactId, CompactContractBytecode> = artifacts
+        let mut report = CoverageReport::default();
+
+        // Collect ASTs and sources
+        let mut versioned_asts: HashMap<Version, HashMap<usize, Ast>> = HashMap::new();
+        let mut versioned_sources: HashMap<Version, HashMap<usize, String>> = HashMap::new();
+        for (path, mut source_file, version) in sources.into_sources_with_version() {
+            // Filter out dependencies
+            // NOTE(onbjerg): For some reason this only returns Some for non-library imports... Is
+            // there a better way?
+            if project_paths.resolve_library_import(std::path::Path::new(&path)).is_none() {
```

Comment:
is `path` here absolute or relative to project dir?

I can add a util function that checks if a path is part of `libs`

Reply:
It seems to be relative